src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -178,13 +147,12 @@ public bool Equals(StringSegment other)
         /// <returns><code>true</code> if the current object is equal to the other parameter; otherwise, <code>false</code>.</returns>
         public bool Equals(StringSegment other, StringComparison comparisonType)
         {
-            int textLength = other.Length;
-            if (Length != textLength)
+            if (Length != other.Length)
```

Comment:
It is basically the same code, but the JIT produces better code.

Current dasm (excerpt):
```asm
G_M65313_IG02:
       488D7DE0             lea      rdi, bword ptr [rbp-20H]
       488B17               mov      rdx, gword ptr [rdi]
       8B4F08               mov      ecx, dword ptr [rdi+8]
       8B7F0C               mov      edi, dword ptr [rdi+12]
       448BC7               mov      r8d, edi
       418BF8               mov      edi, r8d
       448BC7               mov      r8d, edi
       443945FC             cmp      dword ptr [rbp-04H], r8d
       7404                 je       SHORT G_M65313_IG03
       33C0                 xor      eax, eax
       EB1A                 jmp      SHORT G_M65313_IG04
```
The `mov r8d, edi` are crazy stuff.

New dasm (excerpt):
```asm
G_M65314_IG02:
       488D7DE0             lea      rdi, bword ptr [rbp-20H]
       488B17               mov      rdx, gword ptr [rdi]
       8B4F08               mov      ecx, dword ptr [rdi+8]
       448B470C             mov      r8d, dword ptr [rdi+12]
       8B7DFC               mov      edi, dword ptr [rbp-04H]
       413BF8               cmp      edi, r8d
       7404                 je       SHORT G_M65314_IG03
       33C0                 xor      eax, eax
       EB1A                 jmp      SHORT G_M65314_IG04
```
And there is no further load for `other.Length` in the following dasm.

Reply:
/cc @AndyAyersMS, may be an example of interest?

These little `mov` dances are strange :)

Reply:
I've been seeing these move chains pop up more and more often. Using dotnet/coreclr#9066 to track for now, though it probably deserves its own issue.